packages/button/src/index.tsx

```diff
@@ -104,6 +166,8 @@ declare module 'styled-components' {
         appearance?: ButtonAppearanceType
         intent?: ButtonIntentType
         isLoading?: boolean
+        icon?: React.Component
```

Comment:
Use `React.ComponentType` instead of `React.Component` - the latter one doesn't take functional components which might be functionality breaker in the future